### PR TITLE
Add rulesEngine support to CDN

### PR DESCRIPTION
**Pull request by hytao**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by kpajdzik — APPROVED** [review]

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/6088 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/6088/commits/7cad4e67f118d80d999291eb17461c6193d9d27d

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2019-04-15/cdn.json [hunk 1]
```diff
@@ -0,0 +1,4344 @@
...
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+
```
Comment (nschonni) [inline specification/cdn/resource-manager/Microsoft.Cdn/stable/2019-04-15/cdn.json]:
```suggestion
      "description": "Defines the parameters for general match conditions",
```

**Comment by dsgouda** [comment]
@hytao did you push an initial commit where you copied over specs from a previous version. If not, please do so, so that I can review the new changes you are introducing in isolation

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2019-04-15/cdn.json (hunk 1 above)
Comment (hytao) [reply specification/cdn/resource-manager/Microsoft.Cdn/stable/2019-04-15/cdn.json]: Hi, I have fixed this typo.

**Comment by hytao** [comment]
@dsgouda , I have push an initial commit for compassion.

**Comment by hytao** [comment]
Hi, I forgot to add links to the old PR in private repo. https://github.com/Azure/azure-rest-api-specs-pr/pull/746
This change has been approved by ARM reviewer.

@kpajdzik was working with me but his OOF today. Could anyone help me get this PR merged?
@dsgouda @nschonni 

Thanks,
Hanyun

**Comment by dsgouda** [comment]
@hytao There are some model validation errors [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=26129) that need to be fixed.

**Comment by hytao** [comment]
Hi @dsgouda @nschonni @kpajdzik  @sergey-shandar  @veronicagg 

I have fixed most of the validation failure. 

Could someone look at the log and tell me how to get rid of those remaining errors?
https://dev.azure.com/azure-sdk/public/_build/results?buildId=26350

Thanks,

**Comment by kpajdzik** [comment]
> Hi @dsgouda @nschonni @kpajdzik @sergey-shandar @veronicagg
> 
> I have fixed most of the validation failure.
> 
> Could someone look at the log and tell me how to get rid of those remaining errors?
> https://dev.azure.com/azure-sdk/public/_build/results?buildId=26350
> 
> Thanks,

We'll merge it as soon as Veronica is online.

**Comment by kpajdzik** [comment]
Thanks @sergey-shandar!

**Pull request merged**